services/gitlab/gitlab-license.tester.js

```diff
@@ -0,0 +1,57 @@
+import { licenseToColor } from '../licenses.js'
+import { createServiceTester } from '../tester.js'
+export const t = await createServiceTester()
+
+const publicDomainLicenseColor = licenseToColor('MIT License')
+const unknownLicenseColor = licenseToColor()
+
+t.create('License')
+  .get('/guoxudong.io/shields-test/licenced-test.json')
+  .expectBadge({
+    label: 'license',
+    message: 'MIT License',
+    color: `${publicDomainLicenseColor}`,
+  })
+
+t.create('License for repo without a license')
+  .get('/guoxudong.io/shields-test/no-license-test.json')
+  .expectBadge({
+    label: 'license',
+    message: 'not specified',
+    color: 'lightgrey',
+  })
+
+t.create('Other license').get('/gitlab-org/gitlab-foss.json').expectBadge({
+  label: 'license',
+  message: 'Other',
+  color: unknownLicenseColor,
+})
+
+t.create('License for unknown repo')
+  .get('/user1/gitlab-does-not-have-this-repo.json')
+  .expectBadge({
+    label: 'license',
+    message: 'repo not found',
+    color: 'red',
+  })
+
+t.create('Mocking License')
+  .get('/group/project.json')
+  .intercept(nock =>
+    nock('https://gitlab.com')
+      .get('/api/v4/projects/group%2Fproject?license=1')
+      .reply(200, {
+        license: {
+          key: 'apache-2.0',
+          name: 'Apache License 2.0',
+          nickname: '',
+          html_url: 'http://choosealicense.com/licenses/apache-2.0/',
+          source_url: '',
+        },
+      })
+  )
+  .expectBadge({
+    label: 'license',
+    message: 'Apache License 2.0',
+    color: unknownLicenseColor,
+  })
```

Comment:
I don't think we necessarily have to remove this test (in part because I realize it bears similarity to the corresponding github tests), but did want to note that it's not necessary for purposes of service/integration tests.

The license renderer is already extensively tested so we're typically content not trying to retest that logic in service tests

Reply:
Yep, I wrote this test more to show the data structure of `licence`, so that the developer can understand what fields are in the structure directly through this test case, without having to actually call it.

Reply:
Understood, and again don't feel like you need to change anything, but my point is that we don't really do that especially not with our service tests. The schema we define in the service class that is used for validating response objects from the upstream APIs is how we typically "document" the structure and relevant fields, along with trying to include links to the upstream documentation.

Compared to other types of tests our service tests are slow and expensive and we typically use them to provide validation against the upstream services we integrate with/to catch when vendors/platforms make breaking changes to their APIs. They can be used to test/do other things, it's just that we typically try to refrain from doing so because of the associated drawbacks.

This is just an fyi/food for thought in case you decide you'd like to work on other badges in the future, GitLab or otherwise